sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala

```diff
@@ -296,9 +296,15 @@ class JDBCSuite extends SparkFunSuite
     // The older versions of spark have this kind of bugs in parquet data source.
     val df1 = sql("SELECT * FROM foobar WHERE NOT (THEID != 2 AND NAME != 'mary')")
```

Comment:
As I leave the comment in https://github.com/apache/spark/pull/10468#discussion_r151900175, the above test doesn't actually test against SPARK-12218 issue.

Reply:
Fixed.